### ISSUE TITLE
Property is never written on inversed side of OneToOne relation properties 

### ISSUE THREAD

**Issue by alexpozzi** [phpstan/phpstan-doctrine#244]
Hello,
First of all thank you for all your work, it's great!

We recently upgraded to the latest `phpstan/phpstan` and `phpstan/phpstan-doctrine` and we started to have some errors reported on  inversed side of OneToOne relationship properties.

For example, I have two entities:
```php
class Signature {
  // ...

  /**
   * @ORM\OneToOne(targetEntity="App\Entity\Author", inversedBy="signature")
   * @ORM\JoinColumn(onDelete="SET NULL")
   *
   * @var Author|null
   */
  private $author;

  // ...
}

class Author {
  // ...

  /**
   * @ORM\OneToOne(targetEntity="App\Entity\Signature", mappedBy="author")
   *
   * @var Signature|null
   */
  private $signature;

  // ...
}
```
Both entities have a constructor and they are both non read-only entities.
When I run phpstan I received the error `Property App\Entity\Author::$signature is never written, only read. 💡 See: https://phpstan.org/developing-extensions/always-read-written-properties`.
Following the rules explained in https://phpstan.org/developing-extensions/always-read-written-properties it's correct that this error is raised but, I was wondering, if this is really a wanted behavior for this particular case.
If it's not, I'd be happy to try to find a fix for it and submit a PR.

`phpstan/phpstan` version: 1.3.0
`phpstan/phpstan-doctrine` version: 1.0.2

**Comment by pscheit-lillydoo** [comment]
did you configure the object manager in the phpstan extension?

**Comment by alexpozzi** [comment]
Yes

**Comment by jwpage** [comment]
I'm seeing the same behaviour here, with a properly configured object manager.

**Comment by leefant** [comment]
Did anybody find a solution or workaround? I experienced the same with one-to-one relation and properly configured object manager.

Btw, It works fine with $id fields of entities which is never written too. But not with one-to-one

**Comment by fliespl** [comment]
Seeing same issue with self referenced parent - child table.

**Comment by Atala** [comment]
Hello,

I have the same problem on the inversed side of a one-to-one, it looks not "pouplated" to PhpStan. Below taskcollection and tasklistitem are 1-1, with the foreign key beign set on tasklisitem (so I am going reverse here). PhpStan thinks that the "reverse call" will always return NULL. 

```
$item = $taskCollection->getTaskListItem();
if (!is_null($item)) {
```

gives

```
  79     Call to function is_null() with AppBundle\Entity\TaskList\Item will  
         always evaluate to false.                                            
         🪪  function.impossibleType  
```

**Comment by jwpage** [comment]
I think I figured this out, at least for me. For me it was not an issue with `phpstan-doctrine` but rather about how I was constructing the relationship mapping between the entities.

For your example, your constructors would need to look something like this:

```php
class Signature {
  // ...

  /**
   * @ORM\OneToOne(targetEntity="App\Entity\Author", inversedBy="signature")
   * @ORM\JoinColumn(onDelete="SET NULL")
   *
   * @var Author|null
   */
  private $author;

  // ...

  public function __construct(Author $author)
  {
    $this->author = $author;
    $this->author->setSignature($this);
  }
}

class Author {
  // ...

  /**
   * @ORM\OneToOne(targetEntity="App\Entity\Signature", mappedBy="author")
   *
   * @var Signature|null
   */
  private $signature;

  // ...

  public function __construct()
  {

  }

  public function setSignature(Signature $signature): void
  {
    $this->signature = $signature;
  }
}

$author = new Author;
$signature = new Signature($author);
// ...
```

This way when you create the signature for the author (I'm enforcing that signature must have an author in the constructor) you're setting the reference back to the original signature entity so that your entity manager's state is accurate without necessarily re-reading from the database.

Or have I missed the point entirely?
